third_party/move/move-vm/runtime/src/storage/module_storage.rs

```diff
@@ -618,4 +618,13 @@ impl FunctionValueExtension for FunctionValueExtensionAdapter<'_> {
             },
         }
     }
+
+    fn max_value_nest_depth(&self) -> Option<usize> {
+        // TODO: use function value flag as well
+        self.module_storage
+            .runtime_environment()
+            .vm_config()
+            .max_value_nest_depth
+            .map(|v| v as usize)
```

Comment:
why not keep the same types?

Reply:
I think we use depth somewhere as usize, and somewhere as u64.... so I just opted for usize to avoid casts in visitors, but the config stores u64 for now, we can probably set them all to u64